server/node/src/main/java/org/kaaproject/kaa/server/admin/services/KaaAdminServiceImpl.java

```diff
@@ -2800,6 +2801,7 @@ private void validateRecordSchema(String avroSchema, boolean isCtl) throws KaaAd
     }
     
     private void validateRecordSchema(Schema schema) throws KaaAdminServiceException {
+        SchemaUtil.compileAvroSchema(schema);
```

Comment:
Please ensure that "validateRecordSchema" is used during all imports: CTL, Notification, Configuration, Logging and Events